caffe2/image/image_input_op.h

```diff
@@ -192,80 +364,55 @@ bool ImageInputOp<Context>::Prefetch() {
   }
   const int channels = color_ ? 3 : 1;
   // Call mutable_data() once to allocate the underlying memory.
-  prefetched_image_.mutable_data<float>();
+  if (gpu_transform_) {
+    // we'll transfer up in int8, then convert later
+    prefetched_image_.mutable_data<uint8_t>();
+  } else {
+    prefetched_image_.mutable_data<float>();
+  }
   prefetched_label_.mutable_data<int>();
-  // TODO(jiayq): Handle this prefetching with a real thread pool. Currently,
-  // with 4 threads we should be able to get a decent sheed for AlexNet type
-  // training already.
+  // Prefetching handled with a thread pool of "decode_threads" threads.
   std::mt19937 meta_randgen(time(nullptr));
   std::vector<std::mt19937> randgen_per_thread;
-  for (int i = 0; i < 4; ++i) {
+  for (int i = 0; i < num_decode_threads_; ++i) {
     randgen_per_thread.emplace_back(meta_randgen());
   }
-  #pragma omp parallel for num_threads(4)
+
   for (int item_id = 0; item_id < batch_size_; ++item_id) {
     std::bernoulli_distribution mirror_this_image(0.5);
-    std::mt19937& randgen = randgen_per_thread[omp_get_thread_num()];
-    float* image_data = prefetched_image_.mutable_data<float>()
-        + crop_ * crop_ * channels * item_id;
-    string key, value;
+    std::mt19937& randgen = randgen_per_thread[num_decode_threads_];
```

Comment:
I think here is a bug, since randgen_per_thread has only num_decode_threads_ elements. My code crashes here.